core/shared/platform/include/platform_api_extension.h

```diff
@@ -1111,6 +1111,451 @@ os_socket_get_broadcast(bh_socket_t socket, bool *is_enabled);
 int
 os_dumps_proc_mem_info(char *out, unsigned int size);
 
+#if WASM_ENABLE_LIBC_WASI != 0 && WASM_ENABLE_UVWASI != 1
+
+#include "wasmtime_ssp.h"
+
+/****************************************************
+ *                     Section 3                    *
+ *                Filesystem support                *
+ ****************************************************/
```

Comment:
Good point, will add a comment. 

Re sockets, character devices I don't think there's much special consideration. I've stuck all these functions under 'filesystem support' since that's their primary use case. The _majority_ of these functions can only be used with file handles. The exceptions are
`os_fstat`, `os_handle_get/set_flags`, `os_handle_get_access_mode`, `os_close`, `os_readv`, `os_writev` and `os_isatty`.

 In the comment, I'll mention that `os_handle` could also refer to a socket/char device and should support such operations if possible. (I've already drafted the Windows implementation for these functions and didn't run into much issue supporting sockets as well as files - just needed to add an internal 'handle_type' enum in `os_handle` to discriminate if the handle refers to a socket or a handle).